### PR TITLE
Mgrp 258 mayflower docs molecules

### PR DESCRIPTION
**Pull request by gleroux02**
This is ready for review. It adds documentation to the molecule section of the styleguide.

### CODE REVIEW

**Comment by gleroux02** [comment]
I still need to update a few states and add notes to the ones that I do. Otherwise this is ready for review.

File: styleguide/source/_patterns/02-molecules/action-seq-list.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+title: DEPRECATED - Action Sequential List
+---
+Description: A section of content on an action page that includes a title and multiple types of content.
+## State: DEPRECATED
+### Notes
+- This style is deprecated as it should be generalized to display anywhere in the site.
+- The action section defines the type of content it displays via the ```path``` variable it should point to another template. The ```data``` variable will feed the template referenced in ```path```.
+### Contains:
```
Comment (gender404) [inline styleguide/source/_patterns/02-molecules/action-seq-list.md:9]: I think we need to figure out how we want to document something that includes a RTE element.  The implementation in PL will never match how a RTE is implemented in a CMS.  Maybe we should just state it includes an RTE and leave out the path and data object?

File: styleguide/source/_patterns/02-molecules/contact-us.md [hunk 2]
```diff
@@ -1,34 +1,45 @@
...
+Description: Displays groups of contact information.
+## State: Alpha
+### Contains:
+- [@molecules/contact-group](/?p=organisms-contact-group)
+### Used in:
+- [@organisms/by-author/sidebar-contact](/?p=organisms-sidebar-contact)
+## Notes:
+- First two contact groups should always be visible on page load.  Additional contact groups should be hidden within the accordion.
+- `{{ accordion }}` is a local twig variable which is set in the molecule twig template when looping through contactUs.groups and hidden is set.
+### Required Variables
```
Comment (gender404) [inline styleguide/source/_patterns/02-molecules/contact-us.md]: Just a note, I'd like to refactor this a bit to remove the "hidden" variable.  Instead the twig file can split the groups array into to for loops first two and everything else.

File: styleguide/source/_patterns/02-molecules/header-contact.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+Description: Displays a set of contact information in the header.
+
+## State: Alpha
+
+### Contains:
+- [@molecules/contact-group](/?p=organisms-contact-group)
+
+### Used in:
+- [@organisms/by-author/sidebar-contact](/?p=organisms-sidebar-contact)
+
```
Comment (gender404) [inline styleguide/source/_patterns/02-molecules/header-contact.md:13]: this is incorrect.  Should be Action Header.

File: styleguide/source/_patterns/02-molecules/header-contact.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
+---
+title: Header contact
+---
+Description: Displays a set of contact information in the header.
+
+## State: Alpha
+
```
Comment (gender404) [inline styleguide/source/_patterns/02-molecules/header-contact.md:7]: Should this be depreciated as well so we can replace the "address" section with a Contact Group now that address exists as an option within Contact Group?

File: styleguide/source/_patterns/02-molecules/header-search.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+~~~
+statsCallout: {
+  pull:
+    NULL or 'left' or 'right'
+  stat:
+    string/required
+  content:
+    string/required
+}
+~~~
```
Comment (gender404) [inline styleguide/source/_patterns/02-molecules/header-search.md:23]: This is completely wrong.  I think this is the md for the Stats Callout not the header search

File: styleguide/source/_patterns/02-molecules/illustrated-link.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+illustratedLink {
+  text:
+    type: string
+  href:
+    type: string
+  type:
+    type: string
+  image:
+    type: string
+}
```
Comment (gender404) [inline styleguide/source/_patterns/02-molecules/illustrated-link.md:20]: missing required fields

File: styleguide/source/_patterns/02-molecules/location-icons.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+---
+title: Location Icons
+---
+Description: A list of icons displayed for a location.
+## State: Alpha
+### Required Variables
```
Comment (gender404) [inline styleguide/source/_patterns/02-molecules/location-icons.md:10]: Used in Action Header - one of the widgets that go in the right section

File: styleguide/source/_patterns/02-molecules/section-links.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+- [@atoms/05-icons/cat-icon](?p=atoms-cat-icon)
+- [@atoms/05-icons/svg-arrow](?p=atoms-svg-arrow)
+### Used in:
+- [@organisms/by-author/sections-three-up](?p=organisms-sections-three-up)
+### Required Variables
+~~~
+sectionLinks: {
+  catIcon: {
+    svg:
+      type: string/path to svg
```
Comment (gender404) [inline styleguide/source/_patterns/02-molecules/section-links.md]: this as been refactored in this sprint to be "icon" instead of "svg"

**Review by gender404 — CHANGES_REQUESTED** [review]

**Comment by gender404** [comment]
Found another items that needs to be changed.  All md file names with ~ in them need to be changed to - in order for them to work (at least on my PC based local).

**Comment by gender404** [comment]
@patrickfweston - Do you think you can complete these changes this week?  Or do you need some help with them?

File: styleguide/source/_patterns/02-molecules/action-seq-list.md (hunk 1 above)
Comment (gleroux02) [reply styleguide/source/_patterns/02-molecules/action-seq-list.md:9]: I agree that PL will never match how it is implemented in a CMS. However, there is work that needs to be done in order better accommodate both systems and something that can I hope we can resolve in https://jira.state.ma.us/browse/MGRP-287. I agree that we can state that it includes a RTE and then link off to that organism.  If and when 287 is worked and the RTE is updated there would also be less documentation to update.

**Pull request merged**